test/testenv/verificationutils.go

```diff
@@ -252,23 +252,68 @@ func VerifyServiceAccountConfiguredOnPod(deployment *Deployment, ns string, podN
 		}
 		logf.Log.Info("Service Account on Pod", "FOUND", restResponse.Spec.ServiceAccount, "EXPECTED", serviceAccount)
 		return strings.Contains(serviceAccount, restResponse.Spec.ServiceAccount)
-	}, deployment.GetTimeout(), PollInterval).Should(gomega.Equal(true))
+	}, ConsistentDuration, ConsistentPollInterval).Should(gomega.Equal(true))
 }
 
 // VerifyIndexFoundOnPod verify index found on a given POD
 func VerifyIndexFoundOnPod(deployment *Deployment, podName string, indexName string) {
-	gomega.Eventually(func() bool {
-		indexFound := GetIndexOnPod(deployment, podName, indexName)
+	gomega.Consistently(func() bool {
+		indexFound, _ := GetIndexOnPod(deployment, podName, indexName)
 		logf.Log.Info("Checking status of index on pod", "PODNAME", podName, "INDEX NAME", indexName, "STATUS", indexFound)
 		return indexFound
-	}, deployment.GetTimeout(), PollInterval).Should(gomega.Equal(true))
+	}, ConsistentDuration, ConsistentPollInterval).Should(gomega.Equal(true))
+}
+
+// VerifyIndexConfigsMatch verify index specific config
+func VerifyIndexConfigsMatch(deployment *Deployment, podName string, indexName string, maxGlobalDataSizeMB int, maxGlobalRawDataSizeMB int) {
+	gomega.Consistently(func() bool {
+		indexFound, data := GetIndexOnPod(deployment, podName, indexName)
+		logf.Log.Info("Checking status of index on pod", "PODNAME", podName, "INDEX NAME", indexName, "STATUS", indexFound)
+		if indexFound == true {
+			if data.Content.MaxGlobalDataSizeMB == maxGlobalDataSizeMB && data.Content.MaxGlobalRawDataSizeMB == int(maxGlobalRawDataSizeMB) {
+				logf.Log.Info("Checking index configs", "MaxGlobalDataSizeMB", data.Content.MaxGlobalDataSizeMB, "MaxGlobalRawDataSizeMB", data.Content.MaxGlobalRawDataSizeMB)
+				return true
+			}
+		}
+		return false
+	}, ConsistentDuration, ConsistentPollInterval).Should(gomega.Equal(true))
 }
 
 // VerifyIndexExistsOnS3 Verify Index Exists on S3
-func VerifyIndexExistsOnS3(deployment *Deployment, podName string, indexName string) {
+func VerifyIndexExistsOnS3(deployment *Deployment, indexName string, podName string) {
 	gomega.Eventually(func() bool {
 		indexFound := CheckPrefixExistsOnS3(indexName)
 		logf.Log.Info("Checking Index on S3", "INDEX NAME", indexName, "STATUS", indexFound)
+		if !indexFound {
+			logf.Log.Info("Index NOT found. Rolling buckets again", "Index Name", indexName)
+			RollHotToWarm(deployment, podName, indexName)
+		}
```

Comment:
Not sure if I am missing the logic here, or I think this condition check is confusing. Better, we have it more predictable way?

Reply:
This is done to avoid false failure. During test I found out that sometimes multiple bucket rolls are required to push data to remote store.